src/main/java/net/minecraftforge/network/ConnectionData.java

```diff
@@ -66,4 +89,86 @@
+            {
+                remoteModVersions = ModList.get().getMods().stream().map(info -> Pair.of(info.getModId(), info.getVersion().toString())).collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+            }
+            return mismatchedChannels.keySet().stream().map(channel -> Pair.of(channel, mismatchedChannels.get(channel).equals(NetworkRegistry.ABSENT) ? NetworkRegistry.ABSENT : remoteModVersions.getOrDefault(channel.getNamespace(), NetworkRegistry.ABSENT))).collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+        }
+
+        public static Map<ResourceLocation, Pair<String, String>> getChannelDataFromSide(Set<ResourceLocation> mismatchedChannelsFilter, ConnectionData connectionData, boolean fromClientSide)
+        {
+            Map<ResourceLocation, String> channelData;
+
+            if (fromClientSide) //mismatch happened on the client, use client data
+            {
+                channelData = NetworkRegistry.buildChannelVersions();
+            }
+            else //mismatch happened on the server, use server data
+            {
+                channelData = connectionData != null ? connectionData.getChannels() : Map.of();
+            }
+            return channelData.keySet().stream().filter(mismatchedChannelsFilter::contains).map(id -> getModDataFromChannel(id, connectionData, fromClientSide)).collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+        }
+
+        public static Pair<ResourceLocation, Pair<String, String>> getModDataFromChannel(ResourceLocation channel, ConnectionData connectionData, boolean fromClientSide)
+        {
+            if (fromClientSide)
+            {
+                return ModList.get().getModContainerById(channel.getNamespace()).map(modContainer -> Pair.of(channel, Pair.of(modContainer.getModInfo().getDisplayName(), modContainer.getModInfo().getVersion().toString()))).orElse(Pair.of(channel, Pair.of(channel.getNamespace(), "")));
+            }
+            else
+            {
+                Map<String, Pair<String, String>> modData = connectionData != null ? connectionData.getModData() : Map.of();
+                return Pair.of(channel, modData.getOrDefault(channel.getNamespace(), Pair.of(channel.getNamespace(), "")));
+            }
+        }
+
+        public static Map<ResourceLocation, Pair<String, String>> getServersideModData(Set<ResourceLocation> mismatchedModsFilter, ConnectionData connectionData)
+        {
+            Map<String, Pair<String, String>> localChannelData = connectionData != null ? connectionData.getModData() : Map.of();
+            Set<String> modIdFilter = mismatchedModsFilter.stream().map(ResourceLocation::getNamespace).collect(Collectors.toSet());
+            return localChannelData.entrySet().stream().filter(e -> modIdFilter.contains(e.getKey())).collect(Collectors.toMap(e -> new ResourceLocation(e.getKey(), ""), Entry::getValue));
+        }
+
+        public boolean containsMismatches()
```

Comment:
This method should be up with the other public methods and should have documentation